src/main/java/dev/cdevents/events/PipelineRunFinishedCDEvent.java

```diff
@@ -0,0 +1,210 @@
+package dev.cdevents.events;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import dev.cdevents.constants.CDEventConstants;
+import dev.cdevents.models.CDEvent;
+import dev.cdevents.models.PipelineRunFinishedSubject;
+
+import java.net.URI;
+
+public class PipelineRunFinishedCDEvent extends CDEvent {
+
+    private static final String CDEVENT_VERSION = "0.1.0";
+    @JsonProperty(required = true)
+    private PipelineRunFinishedSubject subject;
+
+    /**
+     * Constructor to init CDEvent and set the Subject for {@link PipelineRunFinishedCDEvent}.
+     */
+    public PipelineRunFinishedCDEvent() {
+        initCDEvent(currentCDEventType());
+        setSubject(new PipelineRunFinishedSubject(CDEventConstants.SubjectType.PIPELINERUN));
+    }
+
+    /**
+     * @return subject
+     */
+    public PipelineRunFinishedSubject getSubject() {
+        return subject;
+    }
+
+    /**
+     * @param subject
+     */
+    public void setSubject(PipelineRunFinishedSubject subject) {
+        this.subject = subject;
+    }
+
+    /**
+     * @return the current CDEvent type
+     */
+    @Override
+    public String currentCDEventType() {
+        return CDEventConstants.CDEventTypes.PipelineRunFinishedEvent.getEventType().concat(CDEVENT_VERSION);
+    }
+
+    /**
+     * @return the pipeline-run-finished-event schema URL
+     */
+    @Override
+    public String schemaURL() {
+        return String.format("https://cdevents.dev/%s/schema/pipeline-run-finished-event", CDEventConstants.CDEVENTS_SPEC_VERSION);
+    }
+
+    /**
+     * @return the pipeline-run-finished-event schema Json
+     */
+    @Override
+    public String eventSchema() {
```

Comment:
Very nice, thank you!
In future, it would be great to be able to produce this automatically from the schemas in the spec repo.